### ISSUE TITLE
startup of fairly big (160GB) dbase takes 30mins before accessible

### ISSUE THREAD

**Issue by rdelangh** [orientechnologies/orientdb#6657]
## OrientDB Version, operating system, or hardware.
- v2.2.7
## Operating System
- [x] Linux

## Expected behavior and actual behavior
At startup with "dserver.sh" after a clean shutdown, it takes about 30 mins before we can access our 160GB big database from the "console.sh" client program on the local server:

orientdb> use remote:localhost/cdrarch admin adminpwd
Connecting to database [remote:localhost/cdrarch] with user 'admin'...

The logs don't tell me anything usefull about what's going on behind the scenes, but I guess that a kind of consistency check is happening:
...
2016-09-05 22:35:47:850 INFO  [node1473107743544] Adding node 'node1473107743544' in partition: cdrarch db=[*] v=2 [ODistributedDatabaseImpl$1]

This is a startup with "dserver.sh" to try running this database on multiple servers, for the time being there is only this single server active so no communications whatsoever are happening with any other server. I had exactly the same symptom when starting in standalone (using "server.sh").

-> is it not possible to register at shutdown that the consistency is ok, because a clean shutdown is happening? Then during startup, this time-consuming check (if that is what's happening) can be avoided.

**Comment by rdelangh** [comment]
sorry, size is 300 GB meanwhile...

**Comment by rdelangh** [comment]
Next message in the logfile arrived 35 mins later:

2016-09-05 23:10:44:178 INFO  [node1473107743544] Publishing ONLINE status for database node1473107743544.cdrarch... [ODistributedDatabaseImpl]

**Comment by lvca** [comment]
The startup should be immediate, and when you shutdown the server, we save the DBs are fine. Does it happen all the times? Do you have any errors in the log during the shutdown?

**Comment by rdelangh** [comment]
I did a shutdown of this single-node, logs don't say anything worrying : 


2016-09-06 09:19:55:313 WARNI Received signal: SIGTERM [OSignalHandler]
2016-09-06 09:19:55:314 INFO  OrientDB Server is shutting down... [OServer]

At this point, the process is gone.
When I start it up again with "dserver.sh", this time it did startup within a minute ...

I checked the older logs of the previous shutdown/startup (the one that took so long, and for which I launched this topic), and there were also no errors during shutdown; there was a WARNING during startup that the database "GratefulDeadConcerts" was not closed properly, but no messages about my BIG dbase "cdrarch" (and this one took awfully long to open):

shutdown:
2016-09-05 22:33:10:587 WARNI Received signal: SIGTERM [OSignalHandler]
2016-09-05 22:33:10:589 INFO  OrientDB Server is shutting down... [OServer]

startup:
2016-09-05 22:35:40:677 INFO  Loading configuration from: /opt/orientdb/orientdb-community-2.2.7/config/orientdb-server-config.xml... [OServerConfigurationLoaderXml]
2016-09-05 22:35:41:008 INFO  OrientDB Server v2.2.7 (build 2.2.x@rdcab5af4dce4b538bdb4b372abba46e3fc9f19b7; 2016-08-11 15:17:33+0000) is starting up... [OServer]
...
2016-09-05 22:35:47:641 WARNI {db=GratefulDeadConcerts} Storage 'GratefulDeadConcerts' was not closed properly. Will try to recover from write ahead log [OLocalPaginatedStorage]
2016-09-05 22:35:47:641 INFO  {db=GratefulDeadConcerts} Looking for last checkpoint... [OLocalPaginatedStorage]
2016-09-05 22:35:47:644 INFO  {db=GratefulDeadConcerts} Found FUZZY checkpoint. [OLocalPaginatedStorage]
2016-09-05 22:35:47:644 INFO  {db=GratefulDeadConcerts} Data restore procedure from FUZZY checkpoint is started. [OLocalPaginatedStorage]
2016-09-05 22:35:47:646 WARNI {db=GratefulDeadConcerts} Record com.orientechnologies.orient.core.storage.impl.local.paginat
ed.wal.OFuzzyCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=3401324}} will be skipped during data restore [OLocalPaginatedStorage]
2016-09-05 22:35:47:647 WARNI {db=GratefulDeadConcerts} Record OFuzzyCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=3401331}} com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFuzzyCheckpointStartRecord{lsn=null, previousCheckpoint=OLogSequenceNumber{segment=0, position=3401284}} will be skipped during data restore [OLocalPaginatedStorage]
2016-09-05 22:35:47:647 WARNI {db=GratefulDeadConcerts} Record com.orientechnologies.orient.core.storage.impl.local.paginat
ed.wal.OFuzzyCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=3401371}} will be skipped during data restore [OLocalPaginatedStorage]
2016-09-05 22:35:47:648 INFO  {db=GratefulDeadConcerts} Storage data recover was completed [OLocalPaginatedStorage]
...
2016-09-05 22:35:47:825 INFO  [node1473107743544] Opening database 'cdrarch'... [OHazelcastPlugin]
2016-09-05 22:35:47:838 INFO  [node1473107743544] loaded database configuration from disk: /opt/orientdb/orientdb-community-2.2.7/config/default-distributed-db-config.json [OHazelcastPlugin]
2016-09-05 22:35:47:840 INFO  [node1473107743544] New distributed configuration for database: cdrarch (version=1)
...
2016-09-05 22:35:47:841 INFO  [node1473107743544] Saving distributed configuration file for database 'cdrarch' to: /opt/orientdb/databases/cdrarch/distributed-config.json [OHazelcastPlugin]
2016-09-05 22:35:47:843 INFO  [node1473107743544] Current node started as MASTER for database 'cdrarch' [OHazelcastPlugin$4]
2016-09-05 22:35:47:846 INFO  [node1473107743544] New distributed configuration for database: cdrarch (version=1)
...
2016-09-05 22:35:47:850 INFO  [node1473107743544] Adding node 'node1473107743544' in partition: cdrarch db=[*] v=2 [ODistributedDatabaseImpl$1]
2016-09-05 23:10:44:178 INFO  [node1473107743544] Publishing ONLINE status for database node1473107743544.cdrarch... [ODistributedDatabaseImpl]
2016-09-05 23:10:44:217 INFO  [node1473107743544] New distributed configuration for database: cdrarch (version=2)
...

**Comment by laa** [comment]
Hi @rdelangh log is pointing that node was killed but not switched off as result restore process was under the way that is why it consumed 30 minutes to start.

**Comment by rdelangh** [comment]
@laa : 
We only used the bundled "shutdown.sh" script:

Cfr this "shutdown.sh" script:
- it does a KILL command in case the PID-file was found as "bin/orient.pid" ;
- otherwise this command is launched:
"$JAVA" -client $JAVA_OPTS $ORIENTDB_SETTINGS -Dorientdb.config.file="$CONFIG_FILE" \
        -cp "$ORIENTDB_HOME/lib/orientdb-tools-2.2.7.jar:$ORIENTDB_HOME/lib/*" \
        com.orientechnologies.orient.server.OServerShutdownMain $*

Both methods should obviously result in a proper shutdown and closing of the databases.

**Comment by laa** [comment]
@rdelangh seems like they are not . 
Log records like "Storage 'GratefulDeadConcerts' was not closed properly. Will try to recover from write ahead log" is clear indication of that, so we do suggest to use remote shutdown command instead of kill.

**Comment by rdelangh** [comment]
@laa : can you please elaborate on what the "remote shutdown command" is ?
That "kill" is done by the script "shutdown.sh", which has been bundled by the OrientDB developers, so I guess that this method through "kill" is designed to be safe.
If another method, which you describe as "remote shutdown command", is the prescribed method to give more chances on a safe database closure during shutdown, then 
- the documentation should describe this
- the bundled "shutdown.sh" script should have this alternate method instead of a "kill"

**Comment by laa** [comment]
@rdelangh I agree about that, assigned to @robfrank to double check script.

**Comment by rdelangh** [comment]
I did another attempt, "shutdown.sh" the server process, then edited/changed the "distributed-config.json" config file, then restarted the server with "dserver.sh"

Again the startup took about 30 mins before the database is accessible:
...
2016-09-07 10:27:29:113 INFO  [node1473107743544] New distributed configuration for database: cdrarch (version=1)
...
2016-09-07 10:59:57:137 INFO  [node1473107743544] Publishing ONLINE status for database node1473107743544.cdrarch... [ODistributedDatabaseImpl]

Is this as expected after a config change?

**Comment by laa** [comment]
@rdelangh what is content of server log file during this start could you publish it ?

**Comment by rdelangh** [comment]
@laa , between the two above listed log messages, there is only the small table logged which shows the distributed layout of dbase "cdrarch" so far, then 30 mins nothing until the next message.

Do you wish that I copy the entire logfile ?

**Comment by laa** [comment]
Well  no. Do you observe this behavior if you start a server with server.sh ?

**Comment by rdelangh** [comment]
I want to stop the server now, but that is not working:

- the process is running : 
# ps -aef |grep java
root      4276  4274 68 10:27 pts/1    01:59:07 java -server -Xms1024m -Xmx16384m -Djna.nosys=true -XX:+HeapDumpOnOutOfMemoryError -XX:MaxDirectMemorySize=512g -Djava.awt.headless=true -Dfile.encoding=UTF8 -Drhino.opt.level=9 -Dstorage.diskCache.bufferSize=65536 -Djava.util.logging.config.file=/opt/orientdb/orientdb-community-2.2.7/config/orientdb-server-log.properties -Dorientdb.config.file=/opt/orientdb/orientdb-community-2.2.7/config/orientdb-server-config.xml -Dorientdb.www.path=/opt/orientdb/orientdb-community-2.2.7/www -Dorientdb.build.number=2.2.x@rdcab5af4dce4b538bdb4b372abba46e3fc9f19b7; 2016-08-11 15:17:33+0000 -cp /opt/orientdb/orientdb-community-2.2.7/lib/orientdb-server-2.2.7.jar:/opt/orientdb/orientdb-community-2.2.7/lib/*:/opt/orientdb/orientdb-community-2.2.7/plugins/* -Ddistributed=true com.orientechnologies.orient.server.OServerMain

- a "kill -15" (as per the "shutdown.sh" script) only results in a log message that it is shutting down, but it isn't : 
2016-09-07 13:18:39:407 WARNI Received signal: SIGTERM [OSignalHandler]
2016-09-07 13:18:39:409 INFO  OrientDB Server is shutting down... [OServer]

# sleep 60
# ps -aef |grep java
root      4276  4274 96 10:27 pts/1    02:50:42 java -server -Xms1024m -Xmx16384m -Djna.nosys=true -XX:+HeapDumpOnOutOfMemoryError -XX:MaxDirectMemorySize=512g -Djava.awt.headless=true -Dfile.encoding=UTF8 -Drhino.opt.level=9 -D....

The process is consuming CPU time (see the diff in the cpu-time value from the "ps" output). Connections from "console.sh" are not possible anymore. Another "kill -15" has no effect.

-> what's happening ?

**Comment by rdelangh** [comment]
(oops, sorry for this markup effects...)

**Comment by rdelangh** [comment]
Can I do a "kill -9" ? I know how abrupt it is, but the program does not stop on its own, despite several 'gracious' "kill -15" signalling

It is using full 2 out of the 3 installed cores of the box! No more open port in the range 2420..2429. No logs.
-> seems to be hanging in some heavy loop.

Is there a way to dump something of its stack or so, for you to be able to analyze ?

**Comment by laa** [comment]
@rdelangh could you send thread dump so we will know what is going on ?

**Comment by rdelangh** [comment]
@laa, can you please give me instructions about how to send this big dump file ?
Uncompressed, it's about 32GB. The compressing is still going on, but expected to be also a few GB's size...

**Comment by laa** [comment]
@rdelangh you should use ```jstack <pid>``` command. Thread dump should be quite small.

**Comment by rdelangh** [comment]
ok, found a "jstack" as part of openJDK on my server; its output is not so small, after all (463 lines) : 

# ./jstack 4276 |more
4276: Unable to open socket file: target process not responding or HotSpot VM not loaded
The -F option can be used when the target process is not responding
root@orient1:/usr/lib/jvm/jdk1.8.0_92/bin# ./jstack -F 4276 |more
Attaching to process ID 4276, please wait...
Debugger attached successfully.
Server compiler detected.
JVM version is 25.92-b14
Deadlock Detection:

No deadlocks found.

Thread 12726: (state = BLOCKED)
 - sun.misc.Unsafe.park(boolean, long) @bci=0 (Compiled frame; information may be imprecise)
 - java.util.concurrent.locks.LockSupport.parkNanos(java.lang.Object, long) @bci=20, line=215 (Compiled frame)
 - java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(long) @bci=78, line=2078 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take() @bci=124, line=1093 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take() @bci=1, line=809 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.getTask() @bci=149, line=1067 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) @bci=26, line=1127 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor$Worker.run() @bci=5, line=617 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 12623: (state = BLOCKED)
 - java.lang.Shutdown.exit(int) @bci=95, line=212 (Interpreted frame)
 - java.lang.Terminator$1.handle(sun.misc.Signal) @bci=8, line=52 (Interpreted frame)
 - com.orientechnologies.orient.core.OSignalHandler.handle(sun.misc.Signal) @bci=169, line=66 (Interpreted frame)
 - sun.misc.Signal$1.run() @bci=8, line=212 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 12586: (state = BLOCKED)
 - sun.misc.Unsafe.park(boolean, long) @bci=0 (Compiled frame; information may be imprecise)
 - java.util.concurrent.locks.LockSupport.parkNanos(java.lang.Object, long) @bci=20, line=215 (Compiled frame)
 - java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(long) @bci=78, line=2078 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take() @bci=124, line=1093 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take() @bci=1, line=809 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.getTask() @bci=149, line=1067 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) @bci=26, line=1127 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor$Worker.run() @bci=5, line=617 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 12441: (state = BLOCKED)
 - java.lang.Shutdown.exit(int) @bci=95, line=212 (Interpreted frame)
 - java.lang.Terminator$1.handle(sun.misc.Signal) @bci=8, line=52 (Interpreted frame)
 - com.orientechnologies.orient.core.OSignalHandler.handle(sun.misc.Signal) @bci=169, line=66 (Interpreted frame)
 - sun.misc.Signal$1.run() @bci=8, line=212 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 12286: (state = BLOCKED)
 - java.lang.Shutdown.exit(int) @bci=95, line=212 (Interpreted frame)
 - java.lang.Terminator$1.handle(sun.misc.Signal) @bci=8, line=52 (Interpreted frame)
 - com.orientechnologies.orient.core.OSignalHandler.handle(sun.misc.Signal) @bci=169, line=66 (Interpreted frame)
 - sun.misc.Signal$1.run() @bci=8, line=212 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 12114: (state = BLOCKED)
 - java.lang.Shutdown.exit(int) @bci=95, line=212 (Interpreted frame)
 - java.lang.Terminator$1.handle(sun.misc.Signal) @bci=8, line=52 (Interpreted frame)
 - com.orientechnologies.orient.core.OSignalHandler.handle(sun.misc.Signal) @bci=169, line=66 (Interpreted frame)
 - sun.misc.Signal$1.run() @bci=8, line=212 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 12010: (state = BLOCKED)
 - com.orientechnologies.common.concur.lock.OReadersWriterSpinLock.acquireWriteLock() @bci=111, line=135 (Compiled frame; information may be imprecise)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doClose(boolean, boolean) @bci=36, line=3515 (Interpreted frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.close(boolean, boolean) @bci=3, line=459 (Compiled frame)
 - com.orientechnologies.orient.core.storage.OStorageAbstract.shutdown() @bci=3, line=224 (Interpreted frame)
 - com.orientechnologies.orient.core.Orient.shutdownAllStorages() @bci=98, line=405 (Interpreted frame)
 - com.orientechnologies.orient.core.Orient.access$500(com.orientechnologies.orient.core.Orient) @bci=1, line=56 (Interpreted frame)
 - com.orientechnologies.orient.core.Orient$OShutdownEnginesHandler.shutdown() @bci=4, line=925 (Interpreted frame)
 - com.orientechnologies.orient.core.Orient.shutdown() @bci=102, line=331 (Interpreted frame)
 - com.orientechnologies.orient.server.OServer.shutdown() @bci=381, line=492 (Interpreted frame)
 - com.orientechnologies.orient.server.OServerShutdownHook.run() @bci=11, line=39 (Interpreted frame)


Thread 12007: (state = BLOCKED)
 - java.lang.Object.wait(long) @bci=0 (Compiled frame; information may be imprecise)
 - java.lang.Thread.join(long) @bci=38, line=1245 (Interpreted frame)
 - java.lang.Thread.join() @bci=2, line=1319 (Interpreted frame)
 - java.lang.ApplicationShutdownHooks.runHooks() @bci=86, line=106 (Interpreted frame)
 - java.lang.ApplicationShutdownHooks$1.run() @bci=0, line=46 (Interpreted frame)
 - java.lang.Shutdown.runHooks() @bci=39, line=123 (Interpreted frame)
 - java.lang.Shutdown.sequence() @bci=26, line=167 (Interpreted frame)
 - java.lang.Shutdown.exit(int) @bci=95, line=212 (Interpreted frame)
 - java.lang.Terminator$1.handle(sun.misc.Signal) @bci=8, line=52 (Interpreted frame)
 - com.orientechnologies.orient.core.OSignalHandler.handle(sun.misc.Signal) @bci=169, line=66 (Interpreted frame)
 - sun.misc.Signal$1.run() @bci=8, line=212 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 11954: (state = BLOCKED)
 - java.lang.Thread.currentThread() @bci=0 (Compiled frame; information may be imprecise)
 - com.orientechnologies.common.concur.lock.OReadersWriterSpinLock.acquireReadLock() @bci=84, line=77 (Interpreted frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.isClosed() @bci=4, line=438 (Interpreted frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.isClosed() @bci=13, line=1199 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.isClosed() @bci=14, line=1328 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.close() @bci=12, line=1203 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=843, line=214 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11952: (state = BLOCKED)
 - com.orientechnologies.orient.core.storage.impl.local.paginated.atomicoperations.OAtomicOperationsManager.addThreadInWaitingList(java.lang.Thread) @bci=0, line=243 (Compiled frame)
 - com.orientechnologies.orient.core.storage.impl.local.paginated.atomicoperations.OAtomicOperationsManager.startAtomicOperation(java.lang.String, boolean) @bci=90, line=192 (Compiled frame
)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.startStorageTx(com.orientechnologies.orient.core.tx.OTransaction) @bci=86, line=3040 (Compiled frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=465, line=1379 (Compiled
 frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=19, line=1208 (Interpreted fra
me)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedTransactionManager$1.call() @bci=15, line=114 (Compiled frame)
 - com.orientechnologies.orient.core.db.OScenarioThreadLocal.executeAsDistributed(java.util.concurrent.Callable) @bci=24, line=70 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedTransactionManager.commit(com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx, com.orientechnologies.orient
.core.tx.OTransaction, java.lang.Runnable, com.orientechnologies.orient.server.distributed.impl.ODistributedStorageEventListener) @bci=175, line=111 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=100, line=1230 (Interpreted fr
ame)
 - com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit() @bci=151, line=566 (Compiled frame)
 - com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(boolean) @bci=51, line=106 (Compiled frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(boolean) @bci=179, line=2733 (Compiled frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit() @bci=2, line=2703 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=765, line=201 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11950: (state = BLOCKED)
 - java.util.concurrent.ConcurrentLinkedQueue.offer(java.lang.Object) @bci=4, line=328 (Compiled frame)
 - java.util.concurrent.ConcurrentLinkedQueue.add(java.lang.Object) @bci=2, line=297 (Compiled frame)
 - com.orientechnologies.common.concur.lock.OReadersWriterSpinLock.acquireReadLock() @bci=87, line=77 (Interpreted frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.isClosed() @bci=4, line=438 (Interpreted frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.isClosed() @bci=13, line=1199 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.isClosed() @bci=14, line=1328 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.close() @bci=12, line=1203 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=843, line=214 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11948: (state = BLOCKED)
 - java.util.concurrent.ConcurrentLinkedQueue.offer(java.lang.Object) @bci=4, line=328 (Compiled frame)
 - java.util.concurrent.ConcurrentLinkedQueue.add(java.lang.Object) @bci=2, line=297 (Compiled frame)
 - com.orientechnologies.common.concur.lock.OReadersWriterSpinLock.acquireReadLock() @bci=87, line=77 (Interpreted frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.isClosed() @bci=4, line=438 (Interpreted frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.isClosed() @bci=13, line=1199 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.isClosed() @bci=14, line=1328 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.close() @bci=12, line=1203 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=843, line=214 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11946: (state = BLOCKED)
 - java.util.concurrent.ConcurrentLinkedQueue.offer(java.lang.Object) @bci=4, line=328 (Compiled frame)
 - java.util.concurrent.ConcurrentLinkedQueue.add(java.lang.Object) @bci=2, line=297 (Compiled frame)
 - com.orientechnologies.common.concur.lock.OReadersWriterSpinLock.acquireReadLock() @bci=87, line=77 (Compiled frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.isClosed() @bci=4, line=438 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.isClosed() @bci=13, line=1199 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.isClosed() @bci=14, line=1328 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.close() @bci=12, line=1203 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=843, line=214 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11944: (state = BLOCKED)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.unlockIndexKeys(java.util.TreeMap) @bci=0, line=4235 (Interpreted frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=931, line=1437 (Compiled
 frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=19, line=1208 (Interpreted fra
me)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedTransactionManager$1.call() @bci=15, line=114 (Compiled frame)
 - com.orientechnologies.orient.core.db.OScenarioThreadLocal.executeAsDistributed(java.util.concurrent.Callable) @bci=24, line=70 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedTransactionManager.commit(com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx, com.orientechnologies.orient
.core.tx.OTransaction, java.lang.Runnable, com.orientechnologies.orient.server.distributed.impl.ODistributedStorageEventListener) @bci=175, line=111 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=100, line=1230 (Interpreted fr
ame)
 - com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit() @bci=151, line=566 (Compiled frame)
 - com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(boolean) @bci=51, line=106 (Compiled frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(boolean) @bci=179, line=2733 (Compiled frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit() @bci=2, line=2703 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=765, line=201 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11942: (state = BLOCKED)
 - com.orientechnologies.orient.core.storage.impl.local.paginated.atomicoperations.OAtomicOperationsManager.addThreadInWaitingList(java.lang.Thread) @bci=0, line=243 (Compiled frame)
 - com.orientechnologies.orient.core.storage.impl.local.paginated.atomicoperations.OAtomicOperationsManager.startAtomicOperation(java.lang.String, boolean) @bci=90, line=192 (Compiled frame
)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.startStorageTx(com.orientechnologies.orient.core.tx.OTransaction) @bci=86, line=3040 (Compiled frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=465, line=1379 (Compiled
 frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=19, line=1208 (Interpreted fra
me)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedTransactionManager$1.call() @bci=15, line=114 (Compiled frame)
 - com.orientechnologies.orient.core.db.OScenarioThreadLocal.executeAsDistributed(java.util.concurrent.Callable) @bci=24, line=70 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedTransactionManager.commit(com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx, com.orientechnologies.orient
.core.tx.OTransaction, java.lang.Runnable, com.orientechnologies.orient.server.distributed.impl.ODistributedStorageEventListener) @bci=175, line=111 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=100, line=1230 (Interpreted fr
ame)
 - com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit() @bci=151, line=566 (Compiled frame)
 - com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(boolean) @bci=51, line=106 (Compiled frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(boolean) @bci=179, line=2733 (Compiled frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit() @bci=2, line=2703 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=765, line=201 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11940: (state = BLOCKED)
 - java.util.concurrent.ConcurrentLinkedQueue.offer(java.lang.Object) @bci=4, line=328 (Compiled frame)
 - java.util.concurrent.ConcurrentLinkedQueue.add(java.lang.Object) @bci=2, line=297 (Compiled frame)
 - com.orientechnologies.common.concur.lock.OReadersWriterSpinLock.acquireReadLock() @bci=87, line=77 (Interpreted frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.isClosed() @bci=4, line=438 (Interpreted frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.isClosed() @bci=13, line=1199 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.isClosed() @bci=14, line=1328 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.close() @bci=12, line=1203 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=843, line=214 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11936: (state = BLOCKED)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=932, line=632 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11915: (state = BLOCKED)
 - java.util.concurrent.ConcurrentLinkedQueue.offer(java.lang.Object) @bci=4, line=328 (Compiled frame)
 - java.util.concurrent.ConcurrentLinkedQueue.add(java.lang.Object) @bci=2, line=297 (Compiled frame)
 - com.orientechnologies.common.concur.lock.OReadersWriterSpinLock.acquireReadLock() @bci=87, line=77 (Interpreted frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.isClosed() @bci=4, line=438 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.isClosed() @bci=13, line=1199 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.isClosed() @bci=14, line=1328 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.close() @bci=12, line=1203 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=843, line=214 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Interpreted frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11912: (state = BLOCKED)
 - com.orientechnologies.orient.core.storage.impl.local.paginated.atomicoperations.OAtomicOperationsManager.addThreadInWaitingList(java.lang.Thread) @bci=0, line=243 (Compiled frame)
 - com.orientechnologies.orient.core.storage.impl.local.paginated.atomicoperations.OAtomicOperationsManager.startAtomicOperation(java.lang.String, boolean) @bci=90, line=192 (Compiled frame
)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.startStorageTx(com.orientechnologies.orient.core.tx.OTransaction) @bci=86, line=3040 (Compiled frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=465, line=1379 (Compiled
 frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=19, line=1208 (Interpreted fra
me)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedTransactionManager$1.call() @bci=15, line=114 (Compiled frame)
 - com.orientechnologies.orient.core.db.OScenarioThreadLocal.executeAsDistributed(java.util.concurrent.Callable) @bci=24, line=70 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedTransactionManager.commit(com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx, com.orientechnologies.orient
.core.tx.OTransaction, java.lang.Runnable, com.orientechnologies.orient.server.distributed.impl.ODistributedStorageEventListener) @bci=175, line=111 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.commit(com.orientechnologies.orient.core.tx.OTransaction, java.lang.Runnable) @bci=100, line=1230 (Interpreted fr
ame)
 - com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit() @bci=151, line=566 (Compiled frame)
 - com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(boolean) @bci=51, line=106 (Compiled frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(boolean) @bci=179, line=2733 (Compiled frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit() @bci=2, line=2703 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=765, line=201 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11904: (state = BLOCKED)
 - java.util.concurrent.ConcurrentLinkedQueue.offer(java.lang.Object) @bci=4, line=328 (Compiled frame)
 - java.util.concurrent.ConcurrentLinkedQueue.add(java.lang.Object) @bci=2, line=297 (Compiled frame)
 - com.orientechnologies.common.concur.lock.OReadersWriterSpinLock.acquireReadLock() @bci=87, line=77 (Interpreted frame)
 - com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.isClosed() @bci=4, line=438 (Compiled frame)
 - com.orientechnologies.orient.server.distributed.impl.ODistributedStorage.isClosed() @bci=13, line=1199 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.isClosed() @bci=14, line=1328 (Interpreted frame)
 - com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.close() @bci=12, line=1203 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(com.orientechnologies.orient.server.network.protocol.http.OHttpRequest, com.oriente
chnologies.orient.server.network.protocol.http.OHttpResponse) @bci=843, line=214 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service() @bci=414, line=165 (Compiled frame)
 - com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute() @bci=587, line=613 (Compiled frame)
 - com.orientechnologies.common.thread.OSoftThread.run() @bci=23, line=77 (Compiled frame)


Thread 11472: (state = BLOCKED)
 - sun.misc.Unsafe.park(boolean, long) @bci=0 (Compiled frame; information may be imprecise)
 - java.util.concurrent.locks.LockSupport.park(java.lang.Object) @bci=14, line=175 (Compiled frame)
 - java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await() @bci=42, line=2039 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take() @bci=24, line=1081 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take() @bci=1, line=809 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.getTask() @bci=149, line=1067 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) @bci=26, line=1127 (Interpreted frame)
 - java.util.concurrent.ThreadPoolExecutor$Worker.run() @bci=5, line=617 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 11471: (state = BLOCKED)
 - java.io.FileOutputStream.writeBytes(byte[], int, int, boolean) @bci=0 (Compiled frame; information may be imprecise)
 - java.io.FileOutputStream.write(byte[], int, int) @bci=8, line=326 (Compiled frame)
 - java.io.BufferedOutputStream.flushBuffer() @bci=20, line=82 (Compiled frame)
 - java.io.BufferedOutputStream.flush() @bci=1, line=140 (Compiled frame)
 - java.io.PrintStream.write(byte[], int, int) @bci=30, line=482 (Compiled frame)
 - sun.nio.cs.StreamEncoder.writeBytes() @bci=120, line=221 (Compiled frame)
 - sun.nio.cs.StreamEncoder.implFlushBuffer() @bci=11, line=291 (Compiled frame)
 - sun.nio.cs.StreamEncoder.flushBuffer() @bci=15, line=104 (Interpreted frame)
 - java.io.OutputStreamWriter.flushBuffer() @bci=4, line=185 (Interpreted frame)
 - java.io.PrintStream.write(java.lang.String) @bci=27, line=527 (Interpreted frame)
 - java.io.PrintStream.print(java.lang.String) @bci=9, line=669 (Interpreted frame)
 - java.io.PrintStream.println(java.lang.Object) @bci=11, line=823 (Interpreted frame)
 - java.lang.Throwable$WrappedPrintStream.println(java.lang.Object) @bci=5, line=748 (Interpreted frame)
 - java.lang.Throwable.printStackTrace(java.lang.Throwable$PrintStreamOrWriter) @bci=84, line=658 (Interpreted frame)
 - java.lang.Throwable.printStackTrace(java.io.PrintStream) @bci=9, line=643 (Interpreted frame)
 - java.lang.ThreadGroup.uncaughtException(java.lang.Thread, java.lang.Throwable) @bci=82, line=1061 (Interpreted frame)
 - java.lang.ThreadGroup.uncaughtException(java.lang.Thread, java.lang.Throwable) @bci=13, line=1052 (Interpreted frame)
 - java.lang.ThreadGroup.uncaughtException(java.lang.Thread, java.lang.Throwable) @bci=13, line=1052 (Interpreted frame)
 - java.lang.Thread.dispatchUncaughtException(java.lang.Throwable) @bci=6, line=1952 (Interpreted frame)


Thread 4280: (state = BLOCKED)


Thread 4514: (state = BLOCKED)
 - java.io.UnixFileSystem.getBooleanAttributes0(java.io.File) @bci=0 (Compiled frame; information may be imprecise)
 - java.io.UnixFileSystem.getBooleanAttributes(java.io.File) @bci=2, line=242 (Compiled frame)
 - java.io.File.exists() @bci=29, line=819 (Compiled frame)
 - com.orientechnologies.orient.core.storage.fs.OFileClassic.open() @bci=8, line=602 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableEntry.makeAcquiredFromClosed(com.orientechnologies.common.collection.closabledictionary.OClosableItem) @bci=27, line=9
8 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer.acquire(java.lang.Object) @bci=60, line=272 (Compiled frame)
 - com.orientechnologies.orient.core.storage.cache.local.OWOWCache$PeriodicalFuzzyCheckpointTask.run() @bci=165, line=1762 (Compiled frame)
 - java.util.concurrent.Executors$RunnableAdapter.call() @bci=4, line=511 (Compiled frame)
 - java.util.concurrent.FutureTask.runAndReset() @bci=47, line=308 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask) @bci=1, line=180 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run() @bci=37, line=294 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) @bci=95, line=1142 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor$Worker.run() @bci=5, line=617 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 4513: (state = BLOCKED)
 - java.io.UnixFileSystem.getSpace(java.io.File, int) @bci=0 (Compiled frame; information may be imprecise)
 - java.io.File.getFreeSpace() @bci=43, line=1844 (Compiled frame)
 - com.orientechnologies.orient.core.storage.impl.local.paginated.wal.ODiskWriteAheadLog.checkFreeSpace() @bci=4, line=936 (Compiled frame)
 - com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OLogSegment$FlushTask.run() @bci=32, line=143 (Compiled frame)
 - java.util.concurrent.Executors$RunnableAdapter.call() @bci=4, line=511 (Compiled frame)
 - java.util.concurrent.FutureTask.runAndReset() @bci=47, line=308 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask) @bci=1, line=180 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run() @bci=37, line=294 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) @bci=95, line=1142 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor$Worker.run() @bci=5, line=617 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 4512: (state = BLOCKED)
 - sun.misc.Unsafe.park(boolean, long) @bci=0 (Compiled frame; information may be imprecise)
 - java.util.concurrent.locks.LockSupport.parkNanos(java.lang.Object, long) @bci=20, line=215 (Compiled frame)
 - java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(long) @bci=78, line=2078 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take() @bci=124, line=1093 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take() @bci=1, line=809 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.getTask() @bci=149, line=1067 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) @bci=26, line=1127 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor$Worker.run() @bci=5, line=617 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 4502: (state = BLOCKED)
 - java.io.UnixFileSystem.getBooleanAttributes0(java.io.File) @bci=0 (Compiled frame; information may be imprecise)
 - java.io.UnixFileSystem.getBooleanAttributes(java.io.File) @bci=2, line=242 (Compiled frame)
 - java.io.File.exists() @bci=29, line=819 (Compiled frame)
 - com.orientechnologies.orient.core.storage.fs.OFileClassic.open() @bci=8, line=602 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableEntry.makeAcquiredFromClosed(com.orientechnologies.common.collection.closabledictionary.OClosableItem) @bci=27, line=9
8 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer.acquire(java.lang.Object) @bci=60, line=272 (Compiled frame)
 - com.orientechnologies.orient.core.storage.cache.local.OWOWCache$PeriodicalFuzzyCheckpointTask.run() @bci=165, line=1762 (Compiled frame)
 - java.util.concurrent.Executors$RunnableAdapter.call() @bci=4, line=511 (Compiled frame)
 - java.util.concurrent.FutureTask.runAndReset() @bci=47, line=308 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask) @bci=1, line=180 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run() @bci=37, line=294 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) @bci=95, line=1142 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor$Worker.run() @bci=5, line=617 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 4501: (state = BLOCKED)
 - sun.misc.Unsafe.park(boolean, long) @bci=0 (Compiled frame; information may be imprecise)
 - java.util.concurrent.locks.LockSupport.park(java.lang.Object) @bci=14, line=175 (Compiled frame)
 - java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await() @bci=42, line=2039 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take() @bci=24, line=1081 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take() @bci=1, line=809 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.getTask() @bci=149, line=1067 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) @bci=26, line=1127 (Interpreted frame)
 - java.util.concurrent.ThreadPoolExecutor$Worker.run() @bci=5, line=617 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 4500: (state = BLOCKED)
 - java.io.PrintStream.write(java.lang.String) @bci=4, line=524 (Interpreted frame)
 - java.io.PrintStream.print(java.lang.String) @bci=9, line=669 (Interpreted frame)
 - java.lang.ThreadGroup.uncaughtException(java.lang.Thread, java.lang.Throwable) @bci=75, line=1059 (Interpreted frame)
 - java.lang.ThreadGroup.uncaughtException(java.lang.Thread, java.lang.Throwable) @bci=13, line=1052 (Interpreted frame)
 - java.lang.ThreadGroup.uncaughtException(java.lang.Thread, java.lang.Throwable) @bci=13, line=1052 (Interpreted frame)
 - java.lang.Thread.dispatchUncaughtException(java.lang.Throwable) @bci=6, line=1952 (Interpreted frame)


Thread 4490: (state = BLOCKED)
 - org.apache.lucene.index.DocumentsWriter.anyChanges() @bci=21, line=315 (Compiled frame; information may be imprecise)
 - org.apache.lucene.index.IndexWriter.nrtIsCurrent(org.apache.lucene.index.SegmentInfos) @bci=23, line=4613 (Compiled frame)
 - org.apache.lucene.index.StandardDirectoryReader.doOpenFromWriter(org.apache.lucene.index.IndexCommit) @bci=18, line=282 (Compiled frame)
 - org.apache.lucene.index.StandardDirectoryReader.doOpenIfChanged(org.apache.lucene.index.IndexCommit) @bci=13, line=261 (Compiled frame)
 - org.apache.lucene.index.StandardDirectoryReader.doOpenIfChanged() @bci=5, line=251 (Compiled frame)
 - org.apache.lucene.index.DirectoryReader.openIfChanged(org.apache.lucene.index.DirectoryReader) @bci=1, line=123 (Compiled frame)
 - org.apache.lucene.search.SearcherManager.refreshIfNeeded(org.apache.lucene.search.IndexSearcher) @bci=49, line=137 (Compiled frame)
 - org.apache.lucene.search.SearcherManager.refreshIfNeeded(java.lang.Object) @bci=5, line=58 (Compiled frame)
 - org.apache.lucene.search.ReferenceManager.doMaybeRefresh() @bci=22, line=176 (Compiled frame)
 - org.apache.lucene.search.ReferenceManager.maybeRefreshBlocking() @bci=14, line=253 (Compiled frame)
 - org.apache.lucene.search.ControlledRealTimeReopenThread.run() @bci=171, line=245 (Interpreted frame)


Thread 4489: (state = BLOCKED)
 - sun.misc.Unsafe.park(boolean, long) @bci=0 (Compiled frame; information may be imprecise)
 - java.util.concurrent.locks.LockSupport.parkNanos(java.lang.Object, long) @bci=20, line=215 (Compiled frame)
 - java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(long) @bci=78, line=2078 (Compiled frame)
 - org.apache.lucene.search.ControlledRealTimeReopenThread.run() @bci=83, line=223 (Interpreted frame)


Thread 4488: (state = BLOCKED)
 - java.util.concurrent.CopyOnWriteArrayList.iterator() @bci=0, line=1076 (Compiled frame)
 - org.apache.lucene.search.ReferenceManager.notifyRefreshListenersRefreshed(boolean) @bci=4, line=284 (Compiled frame)
 - org.apache.lucene.search.ReferenceManager.doMaybeRefresh() @bci=91, line=190 (Compiled frame)
 - org.apache.lucene.search.ReferenceManager.maybeRefreshBlocking() @bci=14, line=253 (Compiled frame)
 - org.apache.lucene.search.ControlledRealTimeReopenThread.run() @bci=171, line=245 (Interpreted frame)


Thread 4389: (state = BLOCKED)
 - java.io.RandomAccessFile.length() @bci=0 (Compiled frame; information may be imprecise)
 - com.orientechnologies.orient.core.storage.fs.OFileClassic.close() @bci=89, line=643 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableEntry.makeClosed() @bci=59, line=168 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer.evict() @bci=73, line=704 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer.access$400(com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer) @bci=1
, line=33 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer$LogOpen.run() @bci=38, line=771 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer.drainWriteBuffer() @bci=27, line=663 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer.drainBuffers() @bci=1, line=614 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer.tryToDrainBuffers() @bci=23, line=603 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer.afterWrite(java.lang.Runnable) @bci=20, line=546 (Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer.logOpen(com.orientechnologies.common.collection.closabledictionary.OClosableEntry) @bci=11, line=507 (
Compiled frame)
 - com.orientechnologies.common.collection.closabledictionary.OClosableLinkedContainer.acquire(java.lang.Object) @bci=108, line=284 (Compiled frame)
 - com.orientechnologies.orient.core.storage.cache.local.OWOWCache$PeriodicalFuzzyCheckpointTask.run() @bci=165, line=1762 (Compiled frame)
 - java.util.concurrent.Executors$RunnableAdapter.call() @bci=4, line=511 (Compiled frame)
 - java.util.concurrent.FutureTask.runAndReset() @bci=47, line=308 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask) @bci=1, line=180 (Compiled frame)
 - java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run() @bci=37, line=294 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) @bci=95, line=1142 (Compiled frame)
 - java.util.concurrent.ThreadPoolExecutor$Worker.run() @bci=5, line=617 (Interpreted frame)
 - java.lang.Thread.run() @bci=11, line=745 (Interpreted frame)


Thread 4335: (state = BLOCKED)
 - sun.misc.Unsafe.park(boolean, long) @bci=0 (Compiled frame; information may be imprecise)
 - java.util.concurrent.locks.LockSupport.park(java.lang.Object) @bci=14, line=175 (Compiled frame)
 - java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt() @bci=1, line=836 (Compiled frame)
 - java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireShared(int) @bci=83, line=967 (Compiled frame)
 - java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireShared(int) @bci=10, line=1283 (Compiled frame)
 - java.util.concurrent.locks.ReentrantReadWriteLock$ReadLock.lock() @bci=5, line=727 (Compiled frame)
 - com.orientechnologies.orient.core.Orient.getStorages() @bci=9, line=689 (Compiled frame)
 - com.orientechnologies.common.profiler.OAbstractProfiler$MemoryChecker.run() @bci=17, line=68 (Interpreted frame)
 - java.util.TimerThread.mainLoop() @bci=221, line=555 (Interpreted frame)
 - java.util.TimerThread.run() @bci=1, line=505 (Interpreted frame)


Thread 4317: (state = BLOCKED)


Thread 4316: (state = BLOCKED)
 - java.lang.Object.wait(long) @bci=0 (Compiled frame; information may be imprecise)
 - java.lang.ref.ReferenceQueue.remove(long) @bci=59, line=143 (Compiled frame)
 - java.lang.ref.ReferenceQueue.remove() @bci=2, line=164 (Compiled frame)
 - java.lang.ref.Finalizer$FinalizerThread.run() @bci=36, line=209 (Compiled frame)


Thread 4315: (state = BLOCKED)
 - java.lang.Object.wait(long) @bci=0 (Compiled frame; information may be imprecise)
 - java.lang.Object.wait() @bci=2, line=502 (Compiled frame)
 - java.lang.ref.Reference.tryHandlePending(boolean) @bci=54, line=191 (Compiled frame)
 - java.lang.ref.Reference$ReferenceHandler.run() @bci=1, line=153 (Compiled frame)

**Comment by laa** [comment]
@rdelangh as I can see you are in the middle of several batch transactions. Is it possible ?

**Comment by rdelangh** [comment]
no, there is no client program running at all ; there is only one 'application' server for the moment from where a client program would be connecting to OrientDB, with one single well-known client program. If we run "ps" on the application host, this well-known client program is not active.
Also, such client program would connect to OrientDB on its port 2480/tcp . If it were doing so, then we must so open network connections on the OrientDB server on this port , but there are none : 

% netstat -an | grep 2480
%

If your analysis is that the server threads would be processing transactions, then it sounds as if these threads are looping, no ?

**Comment by laa** [comment]
@rdelangh could you then create 3 thread dumps of the server with several seconds interval but put thread dumps in gist ))) .

**Comment by rdelangh** [comment]
@laa 
I did a new thread dump now, and all output is exactly the same except one extra thread at the top of the thread list.

**Comment by laa** [comment]
@rdelangh the same like all methods on the same places ?

**Comment by rdelangh** [comment]
@laa yes indeed, exactly the same.

**Comment by laa** [comment]
@rdelangh ok let's try one more time, but please be really attentive, if this time thread dump is the same? 
Could you also send me heap dump in such case ?

**Comment by rdelangh** [comment]
@laa : the output is indeed not 100% the same (according to "diff" on two output files)

I try to take a heap dump using "jmap", but it runs already for more than 2 hours, and the resulting file is already 1.6GB size ...

**Comment by rdelangh** [comment]
The "jmap" heap dump finally terminated, the resulting file is 3.2GB size...

The server process of ODB has also stopped somewhere past night, no logs whatsoever are generated...

Big mystery...

Starting up the program gives following log messages:

...
2016-09-08 08:30:10:062 WARNI {db=cdrarch} Storage 'cdrarch' was not closed properly. Will try to recover from write ahead log [OLocalPaginatedStorage]
2016-09-08 08:30:10:063 INFO  {db=cdrarch} Looking for last checkpoint... [OLocalPaginatedStorage]
2016-09-08 08:30:10:064 INFO  {db=cdrarch} Found FUZZY checkpoint. [OLocalPaginatedStorage]
2016-09-08 08:30:10:064 INFO  {db=cdrarch} Data restore procedure from FUZZY checkpoint is started. [OLocalPaginatedStorage]
2016-09-08 08:30:10:067 WARNI {db=cdrarch} Record OFuzzyCheckpointStartRecord{lsn=OLogSequenceNumber{segment=17575, position=68673438}} com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFuzzyCheckpointStartRecord{lsn=null, previousCheckpoint=OLogSequenceNumber{segment=17575, position=68673391}} will be skipped during data restore [OLocalPaginatedStorage]
2016-09-08 08:30:10:067 WARNI {db=cdrarch} Record OFuzzyCheckpointStartRecord{lsn=OLogSequenceNumber{segment=17575, position=68673478}} com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFuzzyCheckpointStartRecord{lsn=null, previousCheckpoint=OLogSequenceNumber{segment=17575, position=68673438}} will be skipped during data restore [OLocalPaginatedStorage]
2016-09-08 08:30:10:068 WARNI {db=cdrarch} Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFuzzyCheckpointEndRecord{lsn=OLogSequenceNumber{segment=17575, position=68673518}} will be skipped during data restore [OLocalPaginatedStorage]
2016-09-08 08:30:10:069 INFO  {db=cdrarch} Storage data recover was completed [OLocalPaginatedStorage]

This database was now accessible after 4 mins : 


2016-09-08 08:34:41:208 SEVER {db=cdrarch} Cannot restore state of cache for storage placed under /opt/orientdb/databases/cdrarch [O2QCache]
2016-09-08 08:34:42:178 INFO  [node1473107743544] Publishing ONLINE status for database node1473107743544.cdrarch... [ODistributedDatabaseImpl]

-> inconsistent behaviour ... and I am very worried about this shutdown problems !

**Comment by laa** [comment]
Hi @rdelangh could you start a separate issue about close? Meanwhile, I will start work on it from Wednesday, but I need a separate issue to track changes and contact with you.

**Comment by laa** [comment]
Hi @rdelangh , I was puzzled by a thread dump which you sent to me. I concluded that this thread dump is invalid. I explain why. 
All threads in thread dump are blocked by JVM in so-called "safe point" state. It means that either you are taking heap dump at this moment, or you are experiencing memory issues, and GC asked JVM to lock threads to make memory housekeeping. If you encounter a problem that the server is closing for very long time or starting very long time, please create thread dump but without the creation of heap dump at the same moment, also before the creation of thread dump please check GC activity.

**Comment by rdelangh** [comment]
@laa , many thanks for your follow-up.
1) regarding your latest comment : I am not very aware of the methods to take a dump of Java processes, so possibly indeed I did something wrong.
The thread dump I took was using the command "jstack -F $pid"

I you recommend any other command, please thrown it to me by all means ;-)

2) I launched this shutdown discussion in a separate thread, i.e. #6705

**Comment by laa** [comment]
@rdelangh I close this issue in favor of #6705 , I wait for your reply on this issue.
